test/multibuild.spec.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright 2019 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as chai from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import * as fs from 'fs';
+import * as Stream from 'stream';
+
+import { checkExists, fileToTarPack, getDocker } from './build-utils';
+
+import { performBuilds } from '../lib/';
+import BuildMetadata from '../lib/build-metadata';
+import { BuildTask } from '../lib/build-task';
+import { LocalImage } from '../lib/local-image';
+
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+class StreamOutputParser {
+	public outputLines: string[] = [];
+
+	public captureOutput(stream: Stream.Readable) {
+		stream.on('data', data => this.outputLines.push(data.toString()));
+	}
+}
+
+describe('performBuilds()', () => {
+	it.skip('correctly builds a task using build secrets', async () => {
```

Comment:
After writing this test, it's disappointing that I've had to skip it (`it.skip`). I initially forgot that balenaEngine would be required instead of standard Docker. Still, it was useful to run it manually against local balena devices.